src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12939,9 +12939,17 @@ private bool IsPossibleCollectionElement()
 
         private CollectionElementSyntax ParseCollectionElement()
         {
-            return IsAtDotDotToken()
-                ? _syntaxFactory.SpreadElement(this.EatDotDotToken(), this.ParseExpressionCore())
-                : _syntaxFactory.ExpressionElement(this.ParseExpressionCore());
+            if (this.CurrentToken.ContextualKind == SyntaxKind.WithKeyword &&
```

Comment:
Consider adding tests for related `SyntaxFacts` APIs, or take a note to add later:

```
        Assert.Equal(SyntaxKind.ExtensionKeyword, SyntaxFacts.GetContextualKeywordKind("extension"));
        Assert.Equal(SyntaxKind.None, SyntaxFacts.GetKeywordKind("extension"));
        Assert.True(SyntaxFacts.IsContextualKeyword(SyntaxKind.ExtensionKeyword));
        Assert.Equal("extension", SyntaxFacts.GetText(SyntaxKind.ExtensionKeyword));
````

Reply:
will add now.